### PR TITLE
OLD: [BD-46] convert Gatsby documentation website to TypeScript 

### PR DESCRIPTION
**Pull request by PKulkoRaccoonGang**
## Description
Convert Gatsby documentation website to TypeScript.
[JIRA](https://openedx.atlassian.net/browse/PAR-803)

## Merge Checklist

* [ ] If your update includes visual changes, have they been reviewed by a designer? Send them a link to the Netlify deploy preview, if applicable.
* [x] Does your change adhere to the documented [style conventions](https://github.com/openedx/paragon/blob/master/docs/decisions/0012-css-styling-conventions)?
* [ ] Do any prop types have missing descriptions in the Props API tables in the documentation site (check deploy preview)?
* [x] Were your changes tested using all available themes (see theme switcher in the header of the deploy preview, under the "Settings" icon)?
* [ ] Were your changes tested in the `example` app?
* [x] Is there adequate test coverage for your changes?
* [x] Consider whether this change needs to reviewed/QA'ed for accessibility (a11y). If so, please add `wittjeff` and `adamstankiewicz` as reviewers on this PR.

## Post-merge Checklist

* [x] Verify your changes were released to [NPM](https://www.npmjs.com/package/@edx/paragon) at the expected version.
* [x] If you'd like, [share](https://github.com/openedx/paragon/discussions/new?category=show-and-tell) your contribution in [#show-and-tell](https://github.com/openedx/paragon/discussions/categories/show-and-tell).
* [x] 🎉 🙌 Celebrate! Thanks for your contribution.

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @PKulkoRaccoonGang! I've created [BLENDED-1229](https://openedx.atlassian.net/browse/BLENDED-1229) to keep track of it in Jira.   More details are on the [BD-46](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3134685276/BD-46+Paragon+Blended+Support+and+Maintenance) project page.


When this pull request is ready, tag your edX technical lead.

File: package.json [hunk 1]
```diff
@@ -34,20 +34,25 @@
...
+    "@types/react-test-renderer": "^18.0.0",
     "airbnb-prop-types": "^2.12.0",
     "bootstrap": "^4.6.1",
     "classnames": "^2.3.1",
     "email-prop-type": "^3.0.0",
     "font-awesome": "^4.7.0",
     "lodash.uniqby": "^4.7.0",
     "mailto-link": "^1.0.0",
-    "prop-types": "^15.8.1",
+    "prop-types": "^15.7.2",
```
Comment (adamstankiewicz) [inline package.json:55]: [curious] Why is this downgraded?

File: tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    "moduleResolution": "node",
+    "resolveJsonModule": true,
+    "isolatedModules": true,
+    "noEmit": false,
+    "jsx": "react-jsx"
+  },
+  "include": [
+    "src"
+  ]
+}
```
Comment (adamstankiewicz) [inline tsconfig.json:26]: nit: newline

File: www/src/components/Menu.tsx [hunk 3]
```diff
@@ -49,9 +52,15 @@ const menuQuery = graphql`
   }
 `;
 
+export interface ComponentNavItemPropsTypes {
+  id: string,
+  fields: PropTypes.InferProps<{ slug: string }>,
```
Comment (adamstankiewicz) [inline www/src/components/Menu.tsx:57]: [curious] What is `PropTypes.InferProps` here?

**Review by adamstankiewicz — COMMENTED** [review]
Just gave this an early review; looking good! Left a couple clarifying comments. Let me know when this is ready for a final review.

File: www/src/components/Menu.tsx (hunk 3 above)
Comment (PKulkoRaccoonGang) [reply www/src/components/Menu.tsx:57]: With `PropTypes.InferProps` we can extract the type from the PropTypes object

File: package.json (hunk 1 above)
Comment (PKulkoRaccoonGang) [reply package.json:55]: Perhaps this is due to the Node js version

File: tsconfig.json (hunk 2 above)
Comment (PKulkoRaccoonGang) [reply tsconfig.json:26]: Corrected

**Comment by openedx-webhooks** [comment]
@PKulkoRaccoonGang Even though your pull request wasn’t merged, please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/paragon&entry.867055334=https://github.com/openedx/paragon/pull/1278&entry.1484655318=https://github.com/PKulkoRaccoonGang&entry.752974735=2022-05-10+16:46&entry.1917517419=2022-08-05+08:09&entry.2133058324=No) so we can improve your experience in the future.

**Pull request closed without merging**